R/time_binning.R

```diff
@@ -0,0 +1,325 @@
+#' and "point". See Details for a description of each.
+#' @param reps \code{numeric}. A non-negative `numeric` specifying the number of
+#' replications for sampling. This argument is only useful in the case of the "random"
+#' or "point" method being specified in the `method` argument. Defaults to 100.
+#' @param scale \code{character}. Specify the desired geological timescale to be used
+#' "GTS2020" or "GTS2012". This argument is only relevant if "min_ma" and "max_ma"
+#' columns are `character` values of interval names. Available interval names can be
+#' accessed via the call GTS2020$interval_name or GTS2012$interval_name. "GTS2020" is
+#' the default option.
+#' @param return_error \code{logical}. Should a vector of numbers be returned to flag
+#' the rows of the `occdf` that cannot be matched to `character` interval names? This is
+#' only relevant if `occdf$max_ma` and `occdf$min_ma` are `character` values.
+#'
+#' @return A \code{dataframe} of the original input `occdf` with the following appended
+#' columns: occurrence id (`id`), number of bins the occurrence age range covers (`n_bins`), bin assignment
+#' (`bin_assignment`), and bin midpoint (`bin_midpoint`). In the case of the "random" method,
+#' a \code{list} is appended for the bin assignment and midpoint columns. If `method` is
+#' specified as "point", only point age estimates are appended (`point_estimates`) as a
+#' \code{list} in the dataframe.
+#'
+#' @details Five approaches (methods) exist in the `time_binning()` function for
+#' assigning occurrences to time bins:
+#' - Midpoint: The "mid" method is the simplest approach, and uses the midpoint of
+#' the fossil occurrence age range to bin the occurrence.
+#' - Majority: The "majority" method bins an occurrence into the bin which it most overlaps with.
+#' As part of this implementation, the majority percentage overlap of the occurrence is also calculated
+#' and returned as an additional column in `occdf`. If desired, these percentages can be used to further
+#' filter an occurrence dataset.
+#' - All: The "all" method bins an occurrence into every bin its age range covers. For occurrences
+#' with age ranges of more than one bin, the occurrence row is duplicated. Each occurrence is assigned
+#' an ID in the column `occdf$id` so that duplicates can be tracked. Additionally, `occdf$n_bins` records
+#' the number of bins each occurrence appears within.
+#' - Random: The "random" method randomly samples X amount of bins (with replacement) from the bins
+#' that the fossil occurrence age range covers. The `reps` argument determines the number of
+#' times the sample process is repeated. All replications for each occurrence are stored as a list within the
+#' `occdf$bin_assignment` and `occdf$bin_midpoint` columns.
+#' - Point: The "point" method randomly samples X amount of point age estimates from the
+#' age range of the fossil occurrence. Sampling is weighted by a normal probability
```

Comment:
I agree, but I think this is best left for future implementation. I would really like to get feedback from the community before expanding this more (it already has a lot of functionality). I would like to note that the "point" method has also now been updated to sample following a uniform distribution rather than normal. As such, this is now an implementation of @KEichenseer 's suggestion of accounting for bin length. We have also updated the documentation to make this clear. Our preference for the "random" method is to keep it as randomly selecting one of the potential bins, regardless of bin length. However, we have again updated the documentation to reflect this.